### PR TITLE
Much simpler permissions

### PR DESCRIPTION
**Pull request by harshad1**

### CODE REVIEW

**Comment by harshad1** [comment]
This branch is a drastic simplification of the other permission branch (#1957). We simply ask for file permissions when markor starts. Markor cannot be used without file permissions.

All other permission testing has been removed for simplicity. 

There are a lot of other smaller changes which general cleanups / tweaks and are not strictly related. I will be cleaning these up a little. But this branch is generally ready for review.

**Comment by k3b** [comment]
@harshad1 @gsantner  After Merging #1981 with #1983 the app still crashes with procedure described in #1982 .

Here is the crashlog

<pre>
---------------------------- PROCESS STARTED (23744) for package net.gsantner.markor_test ----------------------------
2023-02-18 13:37:53.598 23744-23744 Typeface                net.gsantner.markor_test             I  Current DefaultTypeface ( default_roboto )
2023-02-18 13:37:54.788 23744-23744 MultiDex                net.gsantner.markor_test             I  VM with version 2.1.0 has multidex support
2023-02-18 13:37:54.788 23744-23744 MultiDex                net.gsantner.markor_test             I  Installing application
2023-02-18 13:37:54.788 23744-23744 MultiDex                net.gsantner.markor_test             I  VM has multidex support, MultiDex support library is disabled.
2023-02-18 13:37:54.841 23744-23782 ner.markor_tes          net.gsantner.markor_test             I  Starting a blocking GC Alloc
2023-02-18 13:37:54.841 23744-23782 ner.markor_tes          net.gsantner.markor_test             I  Starting a blocking GC Alloc
2023-02-18 13:37:54.843 23744-23783 ner.markor_tes          net.gsantner.markor_test             I  Waiting for a blocking GC Alloc
2023-02-18 13:37:54.845 23744-23784 ner.markor_tes          net.gsantner.markor_test             I  Waiting for a blocking GC Alloc
2023-02-18 13:37:54.846 23744-23744 ner.markor_tes          net.gsantner.markor_test             I  Waiting for a blocking GC Alloc
2023-02-18 13:37:54.865 23744-23783 ner.markor_tes          net.gsantner.markor_test             I  WaitForGcToComplete blocked Alloc on ClassLinker for 21.793ms
2023-02-18 13:37:54.865 23744-23783 ner.markor_tes          net.gsantner.markor_test             I  Starting a blocking GC Alloc
2023-02-18 13:37:54.865 23744-23784 ner.markor_tes          net.gsantner.markor_test             I  WaitForGcToComplete blocked Alloc on ClassLinker for 20.304ms
2023-02-18 13:37:54.865 23744-23784 ner.markor_tes          net.gsantner.markor_test             I  Starting a blocking GC Alloc
2023-02-18 13:37:54.865 23744-23744 ner.markor_tes          net.gsantner.markor_test             I  WaitForGcToComplete blocked Alloc on ClassLinker for 19.563ms
2023-02-18 13:37:54.865 23744-23744 ner.markor_tes          net.gsantner.markor_test             I  Starting a blocking GC Alloc
2023-02-18 13:37:54.931 23744-23744 WebViewFactory          net.gsantner.markor_test             I  Loading com.google.android.webview version 83.0.4103.106 (code 410410680)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->peekLong(JZ)J (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->pokeLong(JJZ)V (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->pokeInt(JIZ)V (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->peekInt(JZ)I (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->pokeByte(JB)V (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->peekByte(J)B (greylist, reflection, allowed)
2023-02-18 13:37:55.046 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->pokeByteArray(J[BII)V (greylist, reflection, allowed)
2023-02-18 13:37:55.047 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Llibcore/io/Memory;->peekByteArray(J[BII)V (greylist, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->objectFieldOffset(Ljava/lang/reflect/Field;)J (greylist,core-platform-api, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden field Ljava/nio/Buffer;->address:J (greylist, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->arrayBaseOffset(Ljava/lang/Class;)I (greylist,core-platform-api, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->arrayIndexScale(Ljava/lang/Class;)I (greylist, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->getLong(Ljava/lang/Object;J)J (greylist,core-platform-api, reflection, allowed)
2023-02-18 13:37:55.048 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, reflection, allowed)
2023-02-18 13:37:55.049 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->getObject(Ljava/lang/Object;J)Ljava/lang/Object; (greylist, reflection, allowed)
2023-02-18 13:37:55.049 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, reflection, allowed)
2023-02-18 13:37:55.050 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putObject(Ljava/lang/Object;JLjava/lang/Object;)V (greylist, linking, allowed)
2023-02-18 13:37:55.050 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->getInt(Ljava/lang/Object;J)I (greylist, linking, allowed)
2023-02-18 13:37:55.050 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putInt(Ljava/lang/Object;JI)V (greylist, linking, allowed)
2023-02-18 13:37:55.050 23744-23788 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Lsun/misc/Unsafe;->putLong(Ljava/lang/Object;JJ)V (greylist, linking, allowed)
2023-02-18 13:37:55.059 23744-23744 cr_LibraryLoader        net.gsantner.markor_test             I  Loaded native library version number "83.0.4103.106"
2023-02-18 13:37:55.061 23744-23744 cr_CachingUmaRecorder   net.gsantner.markor_test             I  Flushed 3 samples from 3 histograms.
2023-02-18 13:37:55.131 23744-23813 chromium                net.gsantner.markor_test             W  [WARNING:dns_config_service_posix.cc(341)] Failed to read DnsConfig.
2023-02-18 13:37:55.453 23744-23744 PhoneWindow             net.gsantner.markor_test             D  windowLightStatusBar : false, disable View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
2023-02-18 13:37:55.506 23744-23744 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
2023-02-18 13:37:55.507 23744-23744 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
2023-02-18 13:37:56.083 23744-23744 PhoneWindow             net.gsantner.markor_test             D  windowLightStatusBar : false, disable View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
2023-02-18 13:37:56.212 23744-23744 TouchFilte...tageHelper net.gsantner.markor_test             I  [needTouchEventFilter] Displayid : 0
2023-02-18 13:37:56.216 23744-23744 BezellessG...sionFilter net.gsantner.markor_test             D  getdisplaysize, x : 720 y : 1440
2023-02-18 13:37:56.231 23744-23744 TouchFilte...tageHelper net.gsantner.markor_test             I  [needTouchEventFilter] Displayid : 0
2023-02-18 13:37:56.234 23744-23744 BezellessG...sionFilter net.gsantner.markor_test             D  getdisplaysize, x : 720 y : 1440
2023-02-18 13:37:56.288 23744-23831 AdrenoGLES              net.gsantner.markor_test             I  QUALCOMM build                   : d4595eb, I1f9128a98a
	Build Date                       : 06/01/20
	OpenGL ES Shader Compiler Version: EV031.27.05.02
	Local Branch                     : 
	Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.8.6.2.R1.10.00.00.537.053
	Remote Branch                    : NONE
	Reconstruct Branch               : NOTHING
2023-02-18 13:37:56.288 23744-23831 AdrenoGLES              net.gsantner.markor_test             I  Build Config                     : S L 8.0.10 AArch32
2023-02-18 13:37:56.295 23744-23831 AdrenoGLES              net.gsantner.markor_test             I  PFP: 0x00c60400, ME: 0x003ff041
2023-02-18 13:37:56.336 23744-23831 Gralloc3                net.gsantner.markor_test             W  mapper 3.x is not supported
2023-02-18 13:37:56.997 23744-23744 net.gsantn...ewFragment net.gsantner.markor_test             I  newInstance(document=/storage/emulated/0/Documents/mordor/todo.txt, lineNumber=-919385553, preview=false
2023-02-18 13:37:57.120 23744-23744 net.gsantn...ewFragment net.gsantner.markor_test             I  isStateBad testCreateParent /storage/emulated/0/Documents/mordor/todo.txt
2023-02-18 13:37:57.194 23744-23744 net.gsantn...ewFragment net.gsantner.markor_test             I  newInstance(document=/storage/emulated/0/Documents/mordor/QuickNote.md, lineNumber=-919385553, preview=false
2023-02-18 13:37:57.255 23744-23744 net.gsantn...ewFragment net.gsantner.markor_test             I  isStateBad testCreateParent /storage/emulated/0/Documents/mordor/QuickNote.md
2023-02-18 13:37:57.776 23744-23744 View                    net.gsantner.markor_test             W  requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{e1c24d3 V.ED..... ......ID 28,25-143,72} during layout: running second layout pass
2023-02-18 13:37:58.009 23744-23831 OpenGLRenderer          net.gsantner.markor_test             I  Davey! duration=1796ms; Flags=1, IntendedVsync=632781930563549, Vsync=632781947230215, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=632781960459331, AnimationStart=632781960595997, PerformTraversalsStart=632781960612091, DrawStart=632783509582090, SyncQueued=632783551122611, SyncStart=632783553774955, IssueDrawCommandsStart=632783554883497, SwapBuffers=632783727712924, FrameCompleted=632783729257351, DequeueBufferDuration=197000, QueueBufferDuration=410000, 
2023-02-18 13:37:58.024 23744-23744 Choreographer           net.gsantner.markor_test             I  Skipped 105 frames!  The application may be doing too much work on its main thread.
2023-02-18 13:37:58.219 23744-23831 OpenGLRenderer          net.gsantner.markor_test             I  Davey! duration=1957ms; Flags=0, IntendedVsync=632781980692674, Vsync=632783730692604, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=632783745249590, AnimationStart=632783745418653, PerformTraversalsStart=632783931820007, DrawStart=632783932468653, SyncQueued=632783933298496, SyncStart=632783933853028, IssueDrawCommandsStart=632783933972507, SwapBuffers=632783936414538, FrameCompleted=632783939235528, DequeueBufferDuration=280000, QueueBufferDuration=2214000, 
2023-02-18 13:37:58.304 23744-23831 OpenGLRenderer          net.gsantner.markor_test             I  Davey! duration=2043ms; Flags=0, IntendedVsync=632781980692674, Vsync=632783730692604, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=632783745249590, AnimationStart=632783745418653, PerformTraversalsStart=632783931820007, DrawStart=632783958624069, SyncQueued=632783999716101, SyncStart=632784000579903, IssueDrawCommandsStart=632784001106934, SwapBuffers=632784013934278, FrameCompleted=632784025113080, DequeueBufferDuration=175000, QueueBufferDuration=570000, 
2023-02-18 13:37:58.325 23744-23744 TouchFilte...tageHelper net.gsantner.markor_test             I  [needTouchEventFilter] Displayid : 0
2023-02-18 13:37:58.327 23744-23744 BezellessG...sionFilter net.gsantner.markor_test             D  getdisplaysize, x : 720 y : 1440
2023-02-18 13:37:58.560 23744-23814 ner.markor_tes          net.gsantner.markor_test             W  Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (greylist, reflection, allowed)
2023-02-18 13:37:58.569 23744-23814 cr_media                net.gsantner.markor_test             W  Requires BLUETOOTH permission
2023-02-18 13:37:58.577 23744-23744 cr_AwContents           net.gsantner.markor_test             W  WebView.destroy() called while WebView is still attached to window.
2023-02-18 13:37:58.603 23744-23744 cr_AwContents           net.gsantner.markor_test             W  WebView.destroy() called while WebView is still attached to window.
2023-02-18 13:37:58.676 23744-23744 WindowManager           net.gsantner.markor_test             E  android.view.WindowLeaked: Activity net.gsantner.markor.activity.MainActivity has leaked window DecorView@9a88b30[MainActivity] that was originally added here
	at android.view.ViewRootImpl.<init>(ViewRootImpl.java:648)
	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:406)
	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:102)
	at android.app.Dialog.show(Dialog.java:419)
	at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:1009)
	at net.gsantner.opoc.util.GsContextUtils.checkExternalStoragePermission(GsContextUtils.java:2371)
	at net.gsantner.opoc.frontend.settings.GsPermissionChecker.doIfExtStoragePermissionGranted(GsPermissionChecker.java:30)
	at net.gsantner.markor.frontend.settings.MarkorPermissionChecker.doIfExtStoragePermissionGranted(MarkorPermissionChecker.java:27)
	at net.gsantner.markor.activity.MainActivity.onResume(MainActivity.java:209)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1454)
	at android.app.Activity.performResume(Activity.java:7972)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4311)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4353)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2063)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7615)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
2023-02-18 13:37:58.707 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/adpcm
2023-02-18 13:37:58.708 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/x-lg-alac
2023-02-18 13:37:58.709 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/g726
2023-02-18 13:37:58.710 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/x-lg-ms-wma
2023-02-18 13:37:58.711 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/lg-wma-voice
2023-02-18 13:37:58.712 23744-23846 AudioCapabilities       net.gsantner.markor_test             W  Unsupported mime audio/x-lg-flac
2023-02-18 13:37:58.727 23744-23846 VideoCapabilities       net.gsantner.markor_test             W  Unsupported mime video/mjpg
2023-02-18 13:37:58.733 23744-23846 VideoCapabilities       net.gsantner.markor_test             W  Unsupported mime video/theora
2023-02-18 13:37:58.763 23744-23744 cr_AwContents           net.gsantner.markor_test             W  Application attempted to call on a destroyed WebView
	java.lang.Throwable
		at org.chromium.android_webview.AwContents.t(chromium-TrichromeWebViewGoogle.apk-stable-410410680:2)
		at Mt.loadingStateChanged(chromium-TrichromeWebViewGoogle.apk-stable-410410680:2)
		at android.os.MessageQueue.nativePollOnce(Native Method)
		at android.os.MessageQueue.next(MessageQueue.java:336)
		at android.os.Looper.loop(Looper.java:174)
		at android.app.ActivityThread.main(ActivityThread.java:7615)
		at java.lang.reflect.Method.invoke(Native Method)
		at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
2023-02-18 13:37:58.766 23744-23846 Utils                   net.gsantner.markor_test             W  could not parse long range '689-670'
2023-02-18 13:37:58.770 23744-23846 Utils                   net.gsantner.markor_test             W  could not parse long range '689-670'
2023-02-18 13:38:00.672 23744-23744 TouchFilte...tageHelper net.gsantner.markor_test             I  [needTouchEventFilter] Displayid : 0
2023-02-18 13:38:00.676 23744-23744 BezellessG...sionFilter net.gsantner.markor_test             D  getdisplaysize, x : 720 y : 1440
2023-02-18 13:38:03.392 23744-23744 chromium                net.gsantner.markor_test             E  [ERROR:aw_browser_terminator.cc(125)] Renderer process (23792) crash detected (code -1).

</pre>

**Comment by harshad1** [comment]
@k3b That traceback shows several functions which don't exist any more in this branch. Please pull again and try again.

**Comment by k3b** [comment]
@harshad1 i have updated my branch

given that usecase=start after first install:

on my android-10 device the app now crashes in 

<pre>
MainActivity.java
 
 @Override
    public void onSaveInstanceState(final Bundle outState) {
        super.onSaveInstanceState(outState);

        // Save references to fragments
        final FragmentManager manager = getSupportFragmentManager();
        manager.putFragment(outState, Integer.toString(R.id.nav_notebook), _notebook);
        manager.putFragment(outState, Integer.toString(R.id.nav_quicknote), _quicknote);
        manager.putFragment(outState, Integer.toString(R.id.nav_todo), _todo);
        manager.putFragment(outState, Integer.toString(R.id.nav_more), _more);
    }
</pre>

because the fragmens _notebook, _quicknote, ... are all null

I have changed my local code to this

<pre>
    @Override
    public void onSaveInstanceState(final Bundle outState) {
        super.onSaveInstanceState(outState);

        // Save references to fragments
        final FragmentManager manager = getSupportFragmentManager();

        // on first start the fragments may be null because of missing permissions
        if (_notebook != null) manager.putFragment(outState, Integer.toString(R.id.nav_notebook), _notebook);
        if (_quicknote != null) manager.putFragment(outState, Integer.toString(R.id.nav_quicknote), _quicknote);
        if (_todo != null) manager.putFragment(outState, Integer.toString(R.id.nav_todo), _todo);
        if (_more != null) manager.putFragment(outState, Integer.toString(R.id.nav_more), _more);
    }
</pre>

* now the app does not crash any more 
* asks for permissions 
* and then terminates without notice (and without any logcat info ).

after restarting the app all seems to work

**Comment by harshad1** [comment]
@k3b Please try the latest version

**Comment by k3b** [comment]
test result (1st start after install) is still the same as the test on feb19th.

*  the app does not crash any more 👍 
*  asks for permissions
* after granting the permissions **then the app seems to be stoped/gone** 👎 
*  you see the app that was active before the install starts.

----

*seems to be* means you donot see the app and you must call the taskmanager to bring the the main app window (markor-filemanager or markor-tutorial) into forground.

I would expect that after the first app-start/grant-permission that markor-filemanager or markor-tutorial come into foreground if there is no

**Comment by k3b** [comment]
@harshad1 if you want me to test or review some of your code please create a pullrequest from your code to my branch test https://github.com/k3b/markor/tree/test that contains merges of my csv branch changes and your "Much simpler permissions" branch

**Comment by k3b** [comment]
I now start codereview of this pullrequest

File: app/src/main/java/net/gsantner/markor/activity/MainActivity.java
Comment (k3b) [inline app/src/main/java/net/gsantner/markor/activity/MainActivity.java:130]: Antipattern : Silent Error hiding without error logging

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (k3b) [inline app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:758]: why not use /** ... */ instead of // which will allow Androidstudio to show tooltip help when mouse is over the method?

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (k3b) [inline app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:767]: /** ... */

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (k3b) [inline app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:782]: more intentrevealing: rename to joinFilePaths

**Review by k3b — CHANGES_REQUESTED** [review]
I will approve when the remaining issue  "app in background after first start with permission grant" is fixed

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:782]: It's in `GsFileUtils`. IMO the namespace gives us the necessary context. For reference, python's built in equivalent is `os.path.join`

**Comment by harshad1** [comment]
@k3b Could you test this latest version again. If it doesn't work, could you please describe the exact steps to reproduce what you are seeing. I am unable to reproduce any issues locally.

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (gsantner) [reply app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java]: Just join is fine here :-)

File: app/src/main/java/net/gsantner/opoc/frontend/base/GsActivityBase.java
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/frontend/base/GsActivityBase.java:33]:
```suggestion
    protected final AtomicBoolean _activityFirstTimeVisible = new AtomicBoolean(true);
```

File: app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java:780]:
```suggestion
     * Check if a file can be created (parent exists and can be written)
```

**Comment by gsantner** [comment]
Thank you,  
I did a few minor improvements. Now it lets users go through the intro first before immediately asking about permission before showing anything else. Also it properly goes back to Markor now, what happened here after accepting the permission was that I was left without anything of Markor open.

**Comment by harshad1** [comment]
@gsantner @k3b 

This PR may have negatively affected how folders load. This is what I am seeing:
1. When you delete 1 or more files (within markor), the files are removed from the list. They are mearly striked out.
2. When one creates a file, it doesn't show up in the list
3. Both of these issues are resolved by navigating away from and back to the folder (i.e. go to a child / parent and then back)
4. Drag to refresh does not resolve this

**Comment by gsantner** [comment]
From the description on the quick sounds like an issue with the FileWithMetadata cache class + loadFolder

**Comment by harshad1** [comment]
Yeah. I think I see the issue. I will work on a fix tonight.

**Pull request merged**
